### DISCUSSION TITLE
QPainter: how does it draw annotations so that it's not erasing existing content in pdf document

### DISCUSSION THREAD

**Discussion by JustDeny** [JakubMelka/PDF4QT#67] [Q&A] [answered]
Hello. I'm not experienced C++&qt developer. Could you please explain how did you set up qpainter for working(drawing annotations:shapes,lines etc.) with existing pdf file. I tried a long time to figure it out but always got into problem such a initializing QPainter with QPdfWriter and then trying to draw something (for example rectangle) and then it recreating pdf file erasing all content that was in there.

**Comment by JakubMelka** [comment, marked as answer]
"Hello, I apologize for missing this discussion earlier. To paint over a PDF, I start by initializing QPainter with QPdfWriter. I then paint the content onto the painter. After this, QPdfWriter completes the creation of the PDF file. I then open this newly created document and merge its content with the content of the original page.

**Reply by JustDeny** [reply]
Thanks so much
